src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -153,77 +169,156 @@ export default class SimidMapCreative extends BaseSimidCreative {
+      icon: placeIcon
+    });
+    placeMarker.addListener('click', () => {
+      this.closestLocation_ = place.geometry.location;
+      this.displayDirections_(this.currentLocation_, this.closestLocation_);
+    });
+  }
+
+  /**
+   * Displays the route between the starting loaction and destination
+   * based off of the selected travel mode.
+   * @param {!google.maps.LatLng} destination The LatLng coordinates of the destination.
+   * @param {!google.maps.LatLng} startingLocation The LatLng coordinates of the start location.
+   * @private 
+   */
+  displayDirections_(destination, startingLocation) {
+    this.directionsRenderer_.setMap(this.map_);
+    this.calculateRoute_(startingLocation, destination);
+    document.getElementById("travel-method").addEventListener("change", () => {
+      this.calculateRoute_(startingLocation, destination);
+    });
+  }
+
+  /**
+   * Creates a drop down menu where users can choose between
+   * different modes of travel to display directions for.
+   * @private 
+   */
+  createTravelChoices_() {
+    const adContainer = document.getElementById("button_container")
+    const travelMethod = document.createElement('select');
+    travelMethod.setAttribute("id", "travel-method");
+    const walkOption = this.createTravelOption_("Walking");
+    const driveOption = this.createTravelOption_("Driving");
+    const bikeOption = this.createTravelOption_("Bicycling");
+    const transitOption = this.createTravelOption_("Transit");
+    travelMethod.add(driveOption);
+    travelMethod.add(walkOption);
+    travelMethod.add(bikeOption);
+    travelMethod.add(transitOption);
+    travelMethod.classList.add("travel-method");
+    adContainer.append(travelMethod);
+  }
+
+  /**
+   * Creates an option element representing a mode of travel.
+   * @param {!string} travelMode A string representing the
+   * given mode of travel.
+   * @private 
+   */
+  createTravelOption_(travelMode) {
+    const travelOption = document.createElement("option");
+    travelOption.value = travelMode.toUpperCase();
+    travelOption.text = travelMode;
+    return travelOption;
+  }
+
+  /**
+   * Displays the route between the starting loaction and destination
+   * based off of the selected travel mode.
+   * @param {!google.maps.LatLng} start The LatLng coordinates of the start location.
+   * @param {!google.maps.LatLng} end The LatLng coordinates of the end location.
+   * @private 
+   */
+  calculateRoute_(start, end) {
+    const directionsService = new google.maps.DirectionsService();
+    const selectedMode = document.getElementById("travel-method").value;
+    directionsService.route(
+      {
+        origin: start,
+        destination: end,
+        travelMode: [selectedMode]
+      },
+      (response, status) => {
+        if (status == "OK") {
+          this.directionsRenderer_.setDirections(response);
+        } else {
+          window.alert("Directions request failed due to " + status);
```

Comment:
Don't use window.alert (no one likes popups, even informational ones).
Sending a LOG message could be nice, or possibly displaying something in the ad to the user

Reply:
addressed in latest commit